packages/dynamite/dynamite/lib/src/builder/client.dart

```diff
@@ -0,0 +1,651 @@
+import 'package:dynamite/src/type_result/type_result.dart';
+
+List<Spec> generateDynamiteOverrides(final State state) => [
+      Class(
+        (final b) => b
+          ..name = '${state.classPrefix}Response'
+          ..types.addAll([
+            refer('T'),
+            refer('U'),
+          ])
+          ..extend = refer('DynamiteResponse<T, U>')
+          ..constructors.add(
+            Constructor(
+              (final b) => b
+                ..requiredParameters.addAll(
+                  ['data', 'headers'].map(
+                    (final name) => Parameter(
+                      (final b) => b
+                        ..name = name
+                        ..toSuper = true,
+                    ),
+                  ),
+                ),
+            ),
+          )
+          ..methods.add(
+            Method(
+              (final b) => b
+                ..name = 'toString'
+                ..returns = refer('String')
+                ..annotations.add(refer('override'))
+                ..lambda = true
+                ..body = Code(
+                  "'${state.classPrefix}Response(data: \$data, headers: \$headers)'",
+                ),
+            ),
+          ),
+      ),
+      Class(
+        (final b) => b
+          ..name = '${state.classPrefix}ApiException'
+          ..extend = refer('DynamiteApiException')
+          ..constructors.add(
+            Constructor(
+              (final b) => b
+                ..requiredParameters.addAll(
+                  ['statusCode', 'headers', 'body'].map(
+                    (final name) => Parameter(
+                      (final b) => b
+                        ..name = name
+                        ..toSuper = true,
+                    ),
+                  ),
+                ),
+            ),
+          )
+          ..methods.addAll([
+            Method(
+              (final b) => b
+                ..name = 'fromResponse'
+                ..returns = refer('Future<${state.classPrefix}ApiException>')
+                ..static = true
+                ..modifier = MethodModifier.async
+                ..requiredParameters.add(
+                  Parameter(
+                    (final b) => b
+                      ..name = 'response'
+                      ..type = refer('HttpClientResponse'),
+                  ),
+                )
+                ..body = Block.of([
+                  const Code('String body;'),
+                  const Code('try {'),
+                  const Code('body = await response.body;'),
+                  const Code('} on FormatException {'),
+                  const Code("body = 'binary';"),
+                  const Code('}'),
+                  const Code(''),
+                  Code('return ${state.classPrefix}ApiException('),
+                  const Code('response.statusCode,'),
+                  const Code('response.responseHeaders,'),
+                  const Code('body,'),
+                  const Code(');'),
```

Comment:
Can you change this to a single object? This is really not readable :sweat_smile: